src/engine/TransitivePath.h

```diff
@@ -383,4 +496,9 @@ class TransitivePath : public Operation {
   static void copyColumns(const IdTableView<INPUT_WIDTH>& inputTable,
                           IdTableStatic<OUTPUT_WIDTH>& outputTable,
                           size_t inputRow, size_t outputRow, size_t skipCol);
+
+  // A small helper function: Insert the `value` to the set at `map[key]`.
+  // As the sets all have an allocator with memory limit, this construction is a
+  // little bit more involved, so this can be a separate helper function.
+  void insertIntoMap(Map& map, Id key, Id value) const;
```

Comment:
Thank you very much.
I think we should definitely split the two implementations.
I think we can just use basic inheritance:
1. A class `TransitivePathBase` that has all the stuff that the class currently has EXCEPT for the override of `computeResult()` + all the helper functions needed for it.
2. One class TransitivePathGraphblas and one class TransitivePathFallback that both inherit from the base and implement their repsecive `computeResult`s.

3. : The base class has a function `static shared_pr<TransitivePathBase> makeTransitivePath(..... all the args, bool use fallback).
(Trick: only declare it in the header, and implement it in the `.cpp` file, in which you can include both implementations.